protos/flyteidl/event/event.proto

```diff
@@ -150,4 +150,20 @@ message TaskExecutionEvent {
     // that should be recorded regardless of the lack of phase change.
     // The version field should be incremented when metadata changes across the duration of an individual phase.
     uint32 phase_version                  = 12;    
+
+    // Metadata around how a task was executed.
+    TaskExecutionMetadata metadata        = 16;
+}
+
+// Holds metadata around how a task was executed.
+message TaskExecutionMetadata {
+    // Includes the type of machine used for this specific task execution. 
+    enum InstanceType {
```

Comment:
Why is this an enum? Is it because interruptible is a top level field in Task. I think that is fine, 
I would like to add other things here like - TaskType, TaskPluginID
and then a map<string,string> of other data

Reply:
because i specifically wanted to define the set of possible values for instance type in particular :)